OpenLaTeXStudio/EditorModule/src/latexstudio/editor/files/FileChooserService.java

```diff
@@ -102,4 +107,47 @@ public static File getSelectedFile(String extension, String description, DialogT
         }
         return null;
     }
+    
+    /**
+    * Get the selected file with Saving Confirmation
+    * @author WhiteHsu
+    * @param currentFile the current file
+    * @param extention the default saving file extention in the Dialog
+    * @param description the default saving description in the Dialog
+    * @param type Dialog Type to determine different confirmation behaviors
+    * @param fixExtention if allow to change file extention in the Dialog
+    * @return the selected file object
+    */
+    public static File getFileWithConfirmation(File currentFile, String extension, String description, DialogType type, boolean fixExtension) {
+        File file = currentFile;
+        if(file == null || type == DialogType.SAVEAS)
+            file = getSelectedFile(extension, description, type, fixExtension);
+        
+        file = confirmFileSave(file, currentFile);
+        
+        return file;
+    }
+    
+    /**
+    * Confirm file saving and show the confirmation dialog
+    * @author WhiteHsu
+    * @param file the selected file
+    * @param currentFIle current file saved in Editor Top Component
+    * @return the selected file object
+    */
+    private static File confirmFileSave(File file, File currentFile) {
+        int reply = JOptionPane.NO_OPTION;
+        while (file != null && file.exists() && !file.equals(currentFile) && reply == JOptionPane.NO_OPTION) {            

```

Comment:
We should change the [reply == NO_OPTION] to [reply != YES_OPTION]. Currently, if I close the confirmation window with X, it saves the file. In my opinion closing the confirmation dialog (and any other operation other than clicking YES) should result in aborting saving.